src/main/java/com/conveyal/taui/controllers/WorkerController.java

```diff
@@ -185,18 +192,83 @@ private String jsonResponse (Response response, int statusCode, Object object) {
         }
     }
 
+    private static class ExtendedJobStatus extends JobStatus {
+        public RegionalAnalysis regionalAnalysis;
+
+        public ExtendedJobStatus(JobStatus js) {
+            super();
+            this.jobId = js.jobId;
+            this.graphId = js.graphId;
+            this.workerCommit = js.workerCommit;
+            this.total = js.total;
+            this.complete = js.complete;
+            this.incomplete = js.incomplete;
+            this.deliveries = js.deliveries;
+            this.deliveryPass = js.deliveryPass;
+        }
+    }
+
     /**
      * Fetch status of all unfinished jobs as a JSON list.
      */
     private String getAllJobs(Request request, Response response) {
-        return jsonResponse(response, HttpStatus.OK_200, broker.getJobSummary());
+        String accessGroup = request.attribute("accessGroup");
+        if (!AnalysisServerConfig.adminAccessGroup.equals(accessGroup)) {
+            response.status(401);
+            return "You do not have access.";
+        }
+
+        List<ExtendedJobStatus> extendedJobStatuses = new ArrayList<>();
+        Collection<JobStatus> jobStatuses = broker.getJobSummary();
+        for (JobStatus js : jobStatuses) {
+            ExtendedJobStatus extendedJobStatus = new ExtendedJobStatus(js);
+            if (!js.jobId.equals("SUM")) {
+                extendedJobStatus.regionalAnalysis = Persistence.regionalAnalyses
+                        .find(QueryBuilder.start("_id").is(js.jobId).get()).next();
+            }
+            extendedJobStatuses.add(extendedJobStatus);
+        }
+
+        return jsonResponse(response, HttpStatus.OK_200, extendedJobStatuses);
+    }
+
+    private static class ExtendedWorkerObservation extends WorkerObservation {
+        public final List<Bundle> bundles;
```

Comment:
Same observation here as for ExtendedJobStatus: this class could be replaced with a single nullable field on the superclass. Here though I kind of question embedding the bundles directly into the workers - this could mean looking up the exact same bundle information in the database hundreds of times, and including an identical copy of it inside each worker. I guess as long as it's not observed to be a performance bottleneck it doesn't matter, and "hundreds" is not a big number.